reactor-netty-http/src/test/java/reactor/netty/http/HttpProtocolsTests.java

```diff
@@ -987,6 +987,27 @@ private void doTest100ContinueConnection(
 		}
 	}
 
+	@ParameterizedCompatibleCombinationsTest
+	void testProtocolVersion(HttpServer server, HttpClient client) {
+		HttpProtocol[] serverProtocols = server.configuration().protocols();
+		HttpProtocol[] clientProtocols = client.configuration().protocols();
+		String configuredProtocol = (serverProtocols.length == 1 && serverProtocols[0] == HttpProtocol.HTTP11) ||
+				(clientProtocols.length == 1 && clientProtocols[0] == HttpProtocol.HTTP11) ? "HTTP/1.1" : "HTTP/2.0";
+
+		disposableServer =
+				server.handle((req, res) -> res.sendString(Mono.just(req.protocol())))
+				      .bindNow();
+
+		client.port(disposableServer.port())
+		      .get()
+		      .uri("/")
+		      .responseSingle((res, bytes) -> bytes.asString().zipWith(Mono.just(res.version().text())))
```

Comment:
Just wondering if the client API should not have a corresponding `.protocol()` returning the textual representation to mirror the server's API?

Reply:
Let's introduce it, if it is requested. wdyt?